agent/api/task/task.go

```diff
@@ -1505,6 +1545,10 @@ func (task *Task) updateContainerDesiredStatusUnsafe(taskDesiredStatus apitaskst
 	for _, container := range task.Containers {
 		taskDesiredStatusToContainerStatus := apitaskstatus.MapTaskToContainerStatus(taskDesiredStatus, container.GetSteadyStateStatus())
 		if container.GetDesiredStatus() < taskDesiredStatusToContainerStatus {
+			if taskDesiredStatusToContainerStatus == apicontainerstatus.ContainerStopped &&
+				container.IsAutoRestartNonEssentialContainer() {
+				container.DesiredToFullyStopWhenReceivingStopped = true
```

Comment:
needs lock? same for all other places that are setting this..

Reply:
Yeah, will use a Set function with lock instead.